tests/markov/msm/test_tram.py

```diff
@@ -0,0 +1,306 @@
+        dtrajs, biases, ttrajs = tram._unpack_input((arr, arr))
+        assert np.array_equal(dtrajs, arr) and np.array_equal(biases, arr) and len(ttrajs) == 0
+    except IndexError:
+        pytest.fail("IndexError while unpacking input!")
+
+
+def test_tram_estimate():
+    trajs = np.asarray([[0, 1, 1, 1, 1, 2, 2, 1, 0, 0], [1, 2, 3, 2, 2, 1, 0, 1, 2, 2], [2, 1, 2, 3, 2, 3, 3, 4, 3, 3],
+                        [3, 2, 2, 3, 4, 4, 3, 4, 3, 2], [3, 2, 3, 3, 4, 4, 3, 4, 4, 3]])
+    trajs = trajs / 5 * 3 - 1.5
+
+    dtrajs = [np.asarray(i, dtype=np.intc) for i in [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 1, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 1, 0, 1, 1, 1, 1, 1],
+                         [1, 0, 0, 1, 1, 1, 1, 1, 1, 0], [1, 0, 1, 1, 1, 1, 1, 1, 1, 1]]]
+
+    bias_centers = [-1, -0.5, 0.0, 0.5, 1]
+
+    def harmonic(x0, x):
+        return 0.1 * (x - x0) ** 2
+    
+    # construct bias matric using harmonic potentials
+    bias_matrices = [np.zeros((len(traj), len(bias_centers)), dtype=np.float64) for traj in dtrajs]
+    for i, traj in enumerate(trajs):
+        for j, bias_center in enumerate(bias_centers):
+            bias = lambda x, x0=bias_center: harmonic(x0, x)
+            bias_matrices[i][:, j] = bias(traj)
+
+    tram = TRAM(maxiter=100, save_convergence_info=True)
+    state_counts = np.asarray([[10, 0], [9, 1], [4, 6], [3, 7], [1, 9]], dtype=np.intc)
+    transition_counts = np.asarray(
+        [[[9, 0], [0, 0]], [[7, 1], [1, 0]], [[2, 2], [1, 4]], [[1, 1], [2, 5]], [[0, 1], [1, 7]]], dtype=np.intc)
+    tram_input = bindings.TRAM_input(state_counts, transition_counts, dtrajs, bias_matrices)
+
+    tram._tram_estimator = bindings.TRAM(tram_input)
+    tram._run_estimation()
+
+
+def test_tram_fit_fetch():
+    trajs = np.asarray([[0, 1, 1, 1, 1, 2, 2, 1, 0, 0], [1, 2, 3, 2, 2, 1, 0, 1, 2, 2], [2, 1, 2, 3, 2, 3, 3, 4, 3, 3],
+                        [3, 2, 2, 3, 4, 4, 3, 4, 3, 2], [3, 2, 3, 3, 4, 4, 3, 4, 4, 3]])
+    trajs = trajs / 5 * 3 - 1.5
+
+    dtrajs = np.asarray([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 1, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 1, 0, 1, 1, 1, 1, 1],
+                         [1, 0, 0, 1, 1, 1, 1, 1, 1, 0], [1, 0, 1, 1, 1, 1, 1, 1, 1, 1]])
+
+    bias_centers = [-1, -0.5, 0.0, 0.5, 1]
+
+    def harmonic(x0, x):
+        return 0.1 * (x - x0) ** 2
+
+    # construct bias matric using harmonic potentials
+    bias_matrices = np.zeros((len(bias_centers), 10, len(bias_centers)))
+    for i, traj in enumerate(trajs):
+        for j, bias_center in enumerate(bias_centers):
+            bias = lambda x, x0=bias_center: harmonic(x0, x)
+            bias_matrices[i, :, j] = bias(traj)
+
+    tram = TRAM(maxiter=100, connectivity='summed_count_matrix', save_convergence_info=True)
+    tram.fit((dtrajs, bias_matrices))
+
+    # energies are identical. so are count matrices. and transition matrices
+    assert np.allclose(tram.therm_state_energies, [0.15673362, 0.077853, 0.04456354, 0.05706922, 0.11557514])
```

Comment:
is it possible to compute these numbers (therm state energies, markov state energies, statdist, transition matrix) from data as reference instead of giving them explicitly?

Reply:
if not easily possible this is fine too...